src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs

```diff
@@ -33,110 +35,26 @@ public FunctionsController(WebScriptHostManager scriptHostManager, WebHookReceiv
-            var authorizationResult = await AuthorizationLevelAttribute.GetAuthorizationResultAsync(request, secretManager, functionName: function.Name);
-            var authorizationLevel = authorizationResult.AuthorizationLevel;
-            request.SetAuthorizationLevel(authorizationLevel);
-            request.SetProperty(ScriptConstants.AzureFunctionsHttpRequestKeyNameKey, authorizationResult.KeyName);
 
-            return authorizationLevel;
-        }
-
-        private async Task<HttpResponseMessage> ProcessRequestAsync(HttpRequestMessage request, FunctionDescriptor function, CancellationToken cancellationToken)
-        {
-            var httpTrigger = function.GetTriggerAttributeOrNull<HttpTriggerAttribute>();
-            bool isWebHook = !string.IsNullOrEmpty(httpTrigger.WebHookType);
-            var authorizationLevel = request.GetAuthorizationLevel();
-            HttpResponseMessage response = null;
-
-            if (isWebHook)
-            {
-                if (request.HasAuthorizationLevel(AuthorizationLevel.Admin))
-                {
-                    // Admin level requests bypass the WebHook auth pipeline
-                    response = await _scriptHostManager.HandleRequestAsync(function, request, cancellationToken);
-                }
-                else
-                {
-                    // This is a WebHook request so define a delegate for the user function.
-                    // The WebHook Receiver pipeline will first validate the request fully
-                    // then invoke this callback.
-                    Func<HttpRequestMessage, Task<HttpResponseMessage>> invokeFunction = async (req) =>
-                    {
-                        // Reset the content stream before passing the request down to the function
-                        Stream stream = await req.Content.ReadAsStreamAsync();
-                        stream.Seek(0, SeekOrigin.Begin);
-
-                        return await _scriptHostManager.HandleRequestAsync(function, req, cancellationToken);
-                    };
-                    response = await _webHookReceiverManager.HandleRequestAsync(function, request, invokeFunction);
-                }
-            }
-            else
-            {
-                // Authorize
-                if (!request.HasAuthorizationLevel(httpTrigger.AuthLevel))
-                {
-                    return new HttpResponseMessage(HttpStatusCode.Unauthorized);
-                }
-
-                // Not a WebHook request so dispatch directly
-                response = await _scriptHostManager.HandleRequestAsync(function, request, cancellationToken);
-            }
-
-            return response;
+            IFuncExecutor proxyFunctionExecutor = new ProxyFunctionExecutor(this._scriptHostManager, controllerContext.Configuration.DependencyResolver, _webHookReceiverManager);
+            request.Properties.Add(ScriptConstants.AzureProxyFunctionExecutorKey, proxyFunctionExecutor);
```

Comment:
Why do you have to pass this as a request property? Seems like this should just be part of ProxyClientExecutor - why can't it be set up with this to begin with, rather than passing it to the executor this way?

Reply:
ProxyClientExecutor only gets created when there are proxies present and we are about to invoke a proxy. at this point we don't know if we the incoming url will be processed by proxy or not so just save the context in the request property.

Reply:
Ok. Is there a way to only create/add this executor if proxies are enabled or being used?

Reply:
Yes we won't know what we route to is a local function or external url here but we do know if the incoming route is for a proxy or not. I will add a check for function.Metadata.IsProxy here when creating/adding this executor.